x/protorev/keeper/posthandler.go

```diff
@@ -0,0 +1,189 @@
+	// 2. If there is no error, then the cache context is written to the main context with no gas consumed
+	cacheCtx, write := ctx.CacheContext()
+	cacheCtx = cacheCtx.WithGasMeter(sdk.NewInfiniteGasMeter())
+
+	// Check if the protorev posthandler can be executed
+	if err := protoRevDec.ProtoRevKeeper.AnteHandleCheck(cacheCtx); err != nil {
+		return next(ctx, tx, simulate)
+	}
+
+	// Extract all of the pools that were swapped in the tx
+	swappedPools := ExtractSwappedPools(tx)
+	if len(swappedPools) == 0 {
+		return next(ctx, tx, simulate)
+	}
+
+	// Attempt to execute arbitrage trades
+	if err := protoRevDec.ProtoRevKeeper.ProtoRevTrade(cacheCtx, swappedPools); err == nil {
+		write()
+		ctx.EventManager().EmitEvents(cacheCtx.EventManager().Events())
+	} else {
+		ctx.Logger().Error("ProtoRevTrade failed with error", err)
+	}
+
+	return next(ctx, tx, simulate)
+}
+
+// AnteHandleCheck checks if the module is enabled and if the number of routes to be processed per block has been reached.
+func (k Keeper) AnteHandleCheck(ctx sdk.Context) error {
+	// Only execute the posthandler if the module is enabled
+	if enabled, err := k.GetProtoRevEnabled(ctx); err != nil || !enabled {
+		return fmt.Errorf("protorev is not enabled")
+	}
+
+	latestBlockHeight, err := k.GetLatestBlockHeight(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get latest block height")
+	}
+
+	currentRouteCount, err := k.GetRouteCountForBlock(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get current route count")
+	}
+
+	maxRouteCount, err := k.GetMaxRoutesPerBlock(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get max iterable routes per block")
+	}
+
+	// Only execute the posthandler if the number of routes to be processed per block has not been reached
+	blockHeight := uint64(ctx.BlockHeight())
+	if blockHeight == latestBlockHeight {
+		if currentRouteCount >= maxRouteCount {
+			return fmt.Errorf("max route count for block has been reached")
+		}
+	} else {
+		// Reset the current route count
+		k.SetRouteCountForBlock(ctx, 0)
+		k.SetLatestBlockHeight(ctx, blockHeight)
+	}
+
+	return nil
+}
+
+// ProtoRevTrade wraps around the build routes, iterate routes, and execute trade functionality to execute cyclic arbitrage trades
+// if they exist. It returns an error if there was an issue executing any single trade.
+func (k Keeper) ProtoRevTrade(ctx sdk.Context, swappedPools []SwapToBackrun) error {
+	// Get the total number of routes that can be explored
+	numberOfIterableRoutes, err := k.CalcNumberOfIterableRoutes(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Iterate and build arbitrage routes for each pool that was swapped on
+	for index := 0; index < len(swappedPools) && numberOfIterableRoutes > 0; index++ {
+		// Build the routes for the pool that was swapped on
+		routes := k.BuildRoutes(ctx, swappedPools[index].TokenInDenom, swappedPools[index].TokenOutDenom, swappedPools[index].PoolId)
+		numRoutes := uint64(len(routes))
```

Comment:
nit: Be consistent with your naming. 

You have "numRoutes" and "numberOfIterableRoutes". Plus it's not clear what numRoutes represents. 

Maybe "numPossibleRoutes" and "numIterableRoutes" to make distinguishing easier.